### ISSUE TITLE
Gem 1.37.0 failing on chef-client 11 (ruby 1.9)

### ISSUE THREAD

**Issue by pdohertybcov** [DataDog/dogapi-rb#197]
I'm on chef-client 11, which I know is old.  And using version 2.16.1 of the datadog cookbook, which I also know is old.

Running the dd-handler recipe blows up with installing the chef-handler-datadog gem, which includes the dogapi gem as a dependency.

I get the stack trace below.
Was this expected?  If the new gem requires a newer version of ruby, can this get noted in the changelog or requirements?

Thanks!

```
Generated at 2019-11-04 09:13:08 -0800
SyntaxError: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:625: syntax error, unexpected tLABEL, expecting ')'
...e_service_level_objective(type: , name: , description: nil, ...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:625: Can't assign to nil
...pe: , name: , description: nil, tags: nil, thresholds: nil, ...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:625: Can't assign to nil
... , description: nil, tags: nil, thresholds: nil, numerator: ...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:625: Can't assign to nil
...il, tags: nil, thresholds: nil, numerator: nil,
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:625: Can't assign to nil
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:626: Can't assign to nil
...              denominator: nil, monitor_ids: nil, monitor_se...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:626: Can't assign to nil
...minator: nil, monitor_ids: nil, monitor_search: nil, groups:...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:626: Can't assign to nil
..._ids: nil, monitor_search: nil, groups: nil)
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:634: syntax error, unexpected tLABEL, expecting ')'
...service_level_objective(slo_id: , type: , name: nil, descrip...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:634: Can't assign to nil
...ve(slo_id: , type: , name: nil, description: nil, tags: nil,...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:634: Can't assign to nil
... , name: nil, description: nil, tags: nil, thresholds: nil,
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:634: Can't assign to nil
...l, description: nil, tags: nil, thresholds: nil,
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:634: Can't assign to nil
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:635: Can't assign to nil
...                numerator: nil, denominator: nil, monitor_id...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:635: Can't assign to nil
...merator: nil, denominator: nil, monitor_ids: nil, monitor_se...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:635: Can't assign to nil
...minator: nil, monitor_ids: nil, monitor_search: nil,
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:635: Can't assign to nil
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:652: syntax error, unexpected tLABEL, expecting ')'
...ervice_level_objective(slo_ids: nil, query: nil, offset: nil...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:652: Can't assign to nil
...e_level_objective(slo_ids: nil, query: nil, offset: nil, lim...
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:652: Can't assign to nil
...ctive(slo_ids: nil, query: nil, offset: nil, limit: nil)
...                               ^
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi/facade.rb:652: Can't assign to nil
...: nil, query: nil, offset: nil, limit: nil)
...                               ^
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/dogapi-1.37.0/lib/dogapi.rb:2:in `<top (required)>'
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-handler-datadog-0.12.2/lib/chef/handler/datadog.rb:5:in `<top (required)>'
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:130:in `require'
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:130:in `rescue in require'
/opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:35:in `require'
/var/cache/chef/cookbooks/datadog/recipes/dd-handler.rb:40:in `from_file'
/opt/chef/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/chef/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_version.rb:234:in `load_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:166:in `load_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:135:in `block in include_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:134:in `each'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:134:in `include_recipe'
/opt/chef/embedded/apps/chef/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
/var/cache/chef/cookbooks/bcw_datadog/recipes/default.rb:15:in `from_file'
/opt/chef/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/chef/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_version.rb:234:in `load_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:166:in `load_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:135:in `block in include_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:134:in `each'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:134:in `include_recipe'
/opt/chef/embedded/apps/chef/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
/var/cache/chef/cookbooks/argus/recipes/default.rb:119:in `from_file'
/opt/chef/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/chef/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/chef/embedded/apps/chef/lib/chef/cookbook_version.rb:234:in `load_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:166:in `load_recipe'
/opt/chef/embedded/apps/chef/lib/chef/run_context/cookbook_compiler.rb:140:in `block in compile_recipes'
/opt/chef/embedded/apps/chef/lib/chef/run_context/cookbook_compiler.rb:138:in `each'
/opt/chef/embedded/apps/chef/lib/chef/run_context/cookbook_compiler.rb:138:in `compile_recipes'
/opt/chef/embedded/apps/chef/lib/chef/run_context/cookbook_compiler.rb:75:in `compile'
/opt/chef/embedded/apps/chef/lib/chef/run_context.rb:89:in `load'
/opt/chef/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:73:in `setup_run_context'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:265:in `setup_run_context'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:429:in `do_run'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:213:in `block in run'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:207:in `fork'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:207:in `run'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:236:in `run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:338:in `block in run_application'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:327:in `loop'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:327:in `run_application'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:55:in `run'
/opt/chef/embedded/apps/chef/bin/chef-client:26:in `<top (required)>'
/usr/bin/chef-client:38:in `load'
/usr/bin/chef-client:38:in `<main>'
```

**Comment by arghyanator** [comment]
We are also being affected by this. The new lines or features in facade.rb in .37 version are (diff between facade.rb of .36 and facade.rb of .37):
```
97a98,99
>       @service_level_objective_svc = Dogapi::V1::ServiceLevelObjectiveService.new(@api_key, @application_key, silent,
>                                                                                   timeout, @datadog_host)
619a622,673
>     # SERVICE LEVEL OBJECTIVES
>     #
> 
>     def create_service_level_objective(type: , name: , description: nil, tags: nil, thresholds: nil, numerator: nil,
>                                        denominator: nil, monitor_ids: nil, monitor_search: nil, groups: nil)
>       @service_level_objective_svc.create_service_level_objective(type: type, name: name, description: description,
>                                                                   tags: tags, thresholds: thresholds,
>                                                                   numerator: numerator, denominator: denominator,
>                                                                   monitor_ids: monitor_ids,
>                                                                   monitor_search: monitor_search, groups: groups)
>     end
> 
>     def update_service_level_objective(slo_id: , type: , name: nil, description: nil, tags: nil, thresholds: nil,
>                                        numerator: nil, denominator: nil, monitor_ids: nil, monitor_search: nil,
>                                        groups: nil)
>       @service_level_objective_svc.update_service_level_objective(slo_id: slo_id, type: type, name: name,
>                                                                   description: description, tags: tags,
>                                                                   thresholds: thresholds, numerator: numerator,
>                                                                   denominator: denominator, monitor_ids: monitor_ids,
>                                                                   monitor_search: monitor_search, groups: groups)
>     end
> 
>     def get_service_level_objective(slo_id)
>       @service_level_objective_svc.get_service_level_objective(slo_id)
>     end
> 
>     def get_service_level_objective_history(slo_id, from_ts, to_ts)
>       @service_level_objective_svc.get_service_level_objective_history(slo_id, from_ts, to_ts)
>     end
> 
>     def search_service_level_objective(slo_ids: nil, query: nil, offset: nil, limit: nil)
>       @service_level_objective_svc.search_service_level_objective(slo_ids: slo_ids, query: query, offset: offset,
>                                                                   limit: limit)
>     end
> 
>     def can_delete_service_level_objective(slo_ids)
>       @service_level_objective_svc.can_delete_service_level_objective(slo_ids)
>     end
> 
>     def delete_service_level_objective(slo_id)
>       @service_level_objective_svc.delete_service_level_objective(slo_id)
>     end
> 
>     def delete_many_service_level_objective(slo_ids)
>       @service_level_objective_svc.delete_many_service_level_objective(slo_ids)
>     end
> 
>     def delete_timeframes_service_level_objective(ops)
>       @service_level_objective_svc.delete_timeframes_service_level_objective(ops)
>     end
> 
>     #
```

**Comment by nmuesch** [comment]
Hey @pdohertybcov apologies for the inconvenience here. Version 1.37.0 of the library included the new Service Level Objective endpoints which were using syntax from Ruby 2.0+

We've just released 1.37.1 of the client which is the same as 1.37.0 except reverts the SLO feature. This should resolve any initialization issues while we work on a longer term solution. Please let us know if you continue to hit any issues here.

**Comment by nmuesch** [comment]
Going to close this out as this should be resolved with the latest release of the client. Apologies again for any inconvenience here.
